### PR TITLE
Apple Wallet: Send Notification Requests to APNs (#78)

### LINKED ISSUE #78: Apple Wallet:  Send Notification Requests to APNs
https://app.dework.xyz/nation3/app-2?taskId=ea8b89fb-99be-470e-a788-6271c8172b6f

Send notification request to Apple Push Notification service (APNs).

https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/sending_notification_requests_to_apns

### PR DESCRIPTION
**Pull request by aahna-ashina**
Send notification requests to Apple Push Notification service (APNs)

## North Star Metric: Number of Active Citizens

- Current metric: **12.43%** of the 177 citizens voted on the last [proposal](https://vote.nation3.org/#/proposal/0x77d3191a72753781303eb2402a1263f837c77771012f687f1395a7a965e396f1).
- Theory:  Sending Nation3 updates to a citizen's Apple Wallet Pass when a new proposal is published will increase voting activity by at least 20%.

## Dework Task

https://app.dework.xyz/nation3/app-2?taskId=ea8b89fb-99be-470e-a788-6271c8172b6f

## Related GitHub Issue

closes #78

## How Has This Been Tested?

- [x] Status checks pass
- [x] Works on Goerli
- [ ] Works on Mainnet

## Are There Admin Tasks?

- [x] @aahna-ashina SQL migration:  `1.0.0.sql` (`passports-goerli`)
- [x] @aahna-ashina SQL migration:  `1.0.0.sql` (`passports-mainnet`)

## Documentation

- [x] https://github.com/nation3/mobile-passport/pull/79/commits/f670502808d1466f29cead9131c52a6fbbee83d7 Document steps for using `/api/pushLastUpdate`

### CODE REVIEW

**Review by luisivan — APPROVED** [review]

**Comment by aahna-ashina** [comment]
@johnmark13 @xPi2 Pending review

**Comment by aahna-ashina** [comment]
> Screen Shot 2022-09-14 at 3 20 12 PM
> 
> 
> 
> I keep getting this while trying to push the last update @aahna-ashina 

@luisivan Maybe because the Vercel preview for this PR has not been deployed?

**Comment by aahna-ashina** [comment]
image

**Comment by aahna-ashina** [comment]
@xPi2 Is it possible to automatically authorize Vercel deployments for my PRs in this repo?

**Comment by luisivan** [comment]
@aahna-ashina tried now, getting this error:
```
{"error":"Internal Server Error: FetchError: {very long string}\n is not a legal HTTP header value"}
```

Maybe `\n` isn't supposed to be there?

**Comment by aahna-ashina** [comment]
@luisivan Let me check.  I'll try to reproduce on Goerli.

**Comment by luisivan** [comment]
It worked!!

**Comment by aahna-ashina** [comment]
@luisivan If we cannot reach 2 review approvals for this PR, do you have admin access to merge it?

File: server/PUSH_NOTIFICATION.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Push Notifications
+
+Apple does not automatically fetch updated passes, so we need to push a notification to the device first.
+
+These are the steps for pushing a notification to registered passes:
+
+1. Add a "latest update" as a new table row in Supabase: 
+   - Go to the table editor in Subapase (Nation3 → passports- Add a "latest update" as a new table row
```
Comment (johnmark13) [inline server/PUSH_NOTIFICATION.md:8]: Minor, spelling of supabase

File: server/cypress/e2e/api/pushLastUpdate.cy.ts [hunk 2]
```diff
@@ -1,18 +1,18 @@
-describe('/api/pushUpdate', () => {
+describe('/api/pushLastUpdate', () => {
 
   it('error when missing username/password', () => {
     cy.request({
-      url: '/api/pushUpdate',
+      url: '/api/pushLasthUpdate',
```
Comment (johnmark13) [inline server/cypress/e2e/api/pushLastUpdate.cy.ts:5]: Minor, extra `h` in pushLastUpdate

File: server/utils/Passes.ts [hunk 3]
```diff
@@ -185,40 +187,58 @@ export class Passes {
...
+              console.log('supabase then result\n', result)
+              if (result.error) {
+                reject(result.error.message)
+              } else {
+                result.data.map((pushTokenObject: any) => {
+                  const pushToken: string = pushTokenObject['push_token']
+                  console.log('pushToken:', pushToken)
+                  
+                  // Send notification request to Apple Push Notification service (APNs)
+                  const apnProvider: apn.Provider = new apn.Provider({
```
Comment (johnmark13) [inline server/utils/Passes.ts:217]: Minor, I would move this to a standalone method to improve readability and to  allow for easiest testing and any changes that may be needed one day.

**Review by johnmark13 — APPROVED** [review]
This is solid as it is, minor comments which do not need reapproving.

File: server/PUSH_NOTIFICATION.md (hunk 1 above)
Comment (aahna-ashina) [reply server/PUSH_NOTIFICATION.md]: @johnmark13 Fixed:  https://github.com/nation3/mobile-passport/pull/79/commits/8b1855413e8df7e76b51010d4035d1de74bfef1c

File: server/cypress/e2e/api/pushLastUpdate.cy.ts (hunk 2 above)
Comment (aahna-ashina) [reply server/cypress/e2e/api/pushLastUpdate.cy.ts]: @johnmark13 Thanks, fixed:  https://github.com/nation3/mobile-passport/pull/79/commits/0f3b0409fa297f32f38f11f39dab48b85ab679d0

File: server/utils/Passes.ts (hunk 3 above)
Comment (aahna-ashina) [reply server/utils/Passes.ts:217]: @johnmark13 Agreed, I would like to add better testing of this function, as it is currently not fully covered by unit tests.  Added this as a refactoring task here:  https://github.com/nation3/mobile-passport/issues/76

Comment (aahna-ashina) [reply server/utils/Passes.ts:217]: @johnmark13 When you say testing, do you mean mock testing?

**Pull request merged**